docusaurus/docs/reference/ui-extensions.md

```diff
@@ -0,0 +1,407 @@
+---
+id: ui-extensions
+title: UI extensions
+description: Reference for UI extensions
+keywords:
+  - grafana
+  - plugins
+  - documentation
+  - plugin.json
+  - API reference
+  - UI extensions
+sidebar_position: 50
+---
+
+This page describes the UI Extensions API in detail.
+
+## Extension points in Grafana
+
+A list of available extension points within Grafana that can be extended by plugins. All these extension point IDs can be accessed using the [`PluginExtensionPoints` enum exposed by the `@grafana/data`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L121) package.
+
+```typescript
+import { PluginExtensionPoints } from '@grafana/data';
+
+const extensionPointId = PluginExtensionPoints.DashboardPanelMenu;
+```
+
+| Property                          | Type      | Description                                                          |
+| --------------------------------- | --------- | -------------------------------------------------------------------- |
+| **`AlertingAlertingRuleAction`**  | Link      | Extend the alert rule menu with custom actions.                      |
+| **`AlertingHomePage`**            | Component | Extend the alerting home page with custom alert-creation experience. |
+| **`AlertingRecordingRuleAction`** | Link      | Extend the alert rule menu with custom actions.                      |
+| **`AlertInstanceAction`**         | Link      | Extend the alert instances table with custom actions.                |
+| **`CommandPalette`**              | Link      | Extend the command palette with plugin specific actions.             |
```

Comment:
We have custom actions and plugin specific actions, but I'm not sure theres really any difference?

Reply:
Don't know. @mckn?

Reply:
Nope, no difference.

Reply:
would just go with custom actions for everything to be consistent then

Reply:
done